frontend/javascripts/dashboard/dataset/dataset_settings_view.tsx

```diff
@@ -204,22 +204,28 @@ class DatasetSettingsView extends React.PureComponent<PropsWithFormAndRouter, St
       // Retrieve the initial dataset rotation settings from the data source config.
       if (doAllLayersHaveTheSameRotation(dataSource.dataLayers)) {
         const firstLayerTransformations = dataSource.dataLayers[0].coordinateTransformations;
-        let initialDatasetRotationSettings: DatasetRotation;
+        let initialDatasetRotationSettings: DatasetRotationSettings;
         if (
           !firstLayerTransformations ||
           firstLayerTransformations.length !== EXPECTED_TRANSFORMATION_LENGTH
         ) {
-          initialDatasetRotationSettings = {
-            x: 0,
-            y: 0,
-            z: 0,
-          };
+          const nulledSetting = { rotationInDegrees: 0, isMirrored: false };
+          initialDatasetRotationSettings = { x: nulledSetting, y: nulledSetting, z: nulledSetting };
         } else {
           initialDatasetRotationSettings = {
             // First transformation is a translation to the coordinate system origin.
-            x: getRotationFromTransformationIn90DegreeSteps(firstLayerTransformations[1], "x"),
-            y: getRotationFromTransformationIn90DegreeSteps(firstLayerTransformations[2], "y"),
-            z: getRotationFromTransformationIn90DegreeSteps(firstLayerTransformations[3], "z"),
+            x: getRotationSettingsFromTransformationIn90DegreeSteps(
```

Comment:
"rotation settings" now means that this also includes mirroring? it's not super clear but I can't come up with something better.